src/algorithms/tracking/TrackPropagation.cc

```diff
@@ -24,31 +32,77 @@
 #include <cmath>
 #include <exception>
 #include <iterator>
+#include <map>
 #include <optional>
+#include <string>
 #include <tuple>
 #include <typeinfo>
 #include <utility>
+#include <variant>
 
-#include "ActsGeometryProvider.h"
-#include "TrackPropagation.h"
+#include "algorithms/tracking/ActsGeometryProvider.h"
+#include "algorithms/tracking/TrackPropagation.h"
+#include "algorithms/tracking/TrackPropagationConfig.h"
 #include "extensions/spdlog/SpdlogToActs.h"
 
-
 namespace eicrecon {
 
-
-    void TrackPropagation::init(std::shared_ptr<const ActsGeometryProvider> geo_svc,
-                                std::shared_ptr<spdlog::logger> logger) {
-        m_geoSvc = geo_svc;
-        m_log = logger;
-        m_log->trace("Initialized");
+template<typename ...L>
+struct multilambda : L... {
+  using L::operator()...;
+  constexpr multilambda(L...lambda) : L(std::move(lambda))... {}
+};
+
+void TrackPropagation::init(const dd4hep::Detector* detector,
+                            std::shared_ptr<const ActsGeometryProvider> geo_svc,
+                            std::shared_ptr<spdlog::logger> logger) {
+    m_geoSvc = geo_svc;
+    m_log = logger;
+
+    std::map<uint32_t,size_t> system_id_layers;
+
+    multilambda _toDouble = {
+      [](const std::string& v) { return dd4hep::_toDouble(v); },
+      [](const double& v)      { return v; },
+    };
+
+    for (auto& surface_variant: m_cfg.surfaces) {
+      if (std::holds_alternative<CylinderSurfaceConfig>(surface_variant)) {
+        CylinderSurfaceConfig surface = std::get<CylinderSurfaceConfig>(surface_variant);
+        const double rmin = std::visit(_toDouble, surface.rmin) / dd4hep::mm * Acts::UnitConstants::mm;
+        const double zmin = std::visit(_toDouble, surface.zmin) / dd4hep::mm * Acts::UnitConstants::mm;
+        const double zmax = std::visit(_toDouble, surface.zmax) / dd4hep::mm * Acts::UnitConstants::mm;
+        const uint32_t system_id = detector->constant<uint32_t>(surface.id);
+        auto bounds = std::make_shared<Acts::CylinderBounds>(rmin, (zmax-zmin)/2);
+        auto t = Acts::Translation3(Acts::Vector3(0, 0, (zmax+zmin)));
```

Comment:
This is an incorrect offset, which isn't apparent in testing due to the symmetric cylinder surfaces, where zmax+zmin is zero. This should be:
```suggestion
        auto t = Acts::Translation3(Acts::Vector3(0, 0, (zmax+zmin)/2));
```